game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -271,80 +272,110 @@ public Territory getTerritory(final String s) {
+  /**
+   * Returns the shortest route between two territories so that covered territories except the start
+   * and end match the condition or null if no route exists. Doesn't pass in any units so doesn't
+   * check canals or movement costs.
+   *
+   * @param cond condition that covered territories of the route must match
+   */
+  @Nullable
+  public Route getRoute(
+      final Territory start, final Territory end, final Predicate<Territory> cond) {
+    checkNotNull(start);
+    checkNotNull(end);
+    return new RouteFinder(this, Matches.territoryIs(end).or(cond))
+        .findRoute(start, end)
+        .orElse(null);
   }
 
+  /** See {@link #getRouteForUnits(Territory, Territory, Predicate, Collection, PlayerId)}. */
   @Nullable
-  public Route getRouteIgnoreEndValidatingCanals(
-      final Territory t1,
-      final Territory t2,
+  public Route getRouteForUnit(
+      final Territory start,
+      final Territory end,
       final Predicate<Territory> cond,
-      final Collection<Unit> units,
+      final Unit unit,
       final PlayerId player) {
-    checkNotNull(t1);
-    checkNotNull(t2);
-    return new RouteFinder(this, Matches.territoryIs(t2).or(cond), units, player)
-        .findRoute(t1, t2)
-        .orElse(null);
+    return getRouteForUnits(start, end, cond, Set.of(unit), player);
   }
 
   /**
-   * A composite route between two territories. Example set of matches: [Friendly Land, score: 1]
-   * [Enemy Land, score: 2] [Neutral Land, score = 4] With this example set, an 8 length friendly
-   * route is considered equal in score to a 4 length enemy route and a 2 length neutral route. This
-   * is because the friendly route score is 1/2 of the enemy route score and 1/4 of the neutral
-   * route score. Note that you can choose whatever scores you want, and that the matches can mix
-   * and match with each other in any way. (Recommended that you use 2,3,4 as scores, unless you
-   * will allow routes to be much longer under certain conditions) Returns null if there is no route
-   * that exists that matches any of the matches.
+   * Returns the route with the minimum movement cost between two territories so that covered
+   * territories except the start and end match the condition or null if no route exists. Also
+   * validates canals between any of the territories. Since different units can have different
+   * movement costs per territory, it takes the max movement cost for each territory of all the
+   * units when comparing movement costs across different routes.
    *
-   * @param start start territory of the route
-   * @param end end territory of the route
-   * @param matches Map of territory matches for covered territories
-   * @return a composite route between two territories
+   * @param cond condition that covered territories of the route must match
+   * @param units checked against canals and for movement costs
+   * @param player player to check canal ownership
```

Comment:
If this method is specific to canal, should that be baked into the method name? eg, something like: `getRouteForUnitsOverCanal`

Reply:
It used to but at this point its more than that. Its essentially the route method if you know the units you want to move as it checks canals and movement costs.

Reply:
Updated it a little bit. It reads fine to me so if you have further suggestions let me know as sometimes the author is just too close to it.